core/unix/os.c

```diff
@@ -9836,15 +9836,12 @@ os_take_over_all_unknown_threads(dcontext_t *dcontext)
         num_thread_takeover_records = threads_to_signal;
         takeover_dcontext = dcontext;
 
-        /* Signal the other threads. */
-        for (i = 0; i < threads_to_signal; i++) {
-            thread_signal(get_process_id(), records[i].tid, SUSPEND_SIGNAL);
-        }
+        /* Signal the other threads one at a time to avoid contention. */
```

Comment:
Hmm, the commit message and this comment seem to ignore the release of the lock.  Isn't the main benefit here releasing the initexit lock ourselves before doing any signaling, rather than the signaling order?

Since you have a test to measure perf, did you measure releasing the lock, having a loop of signals, and then the current loop of waiting?  Intuition-wise it's unclear whether sitting and waiting for one thread at a time to get the signal and init itself outweighs contention among them once we don't hold the lock.